### PR TITLE
[FIX] Sage charge states and spec refs and general spec ref handling

### PR DESCRIPTION
**Pull request by jpfeuffer**
## Description

more bugs

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/source/QC/FragmentMassError.cpp [hunk 1]
```diff
@@ -97,7 +97,7 @@ namespace OpenMS
     {
       throw Exception::InvalidParameter(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No spectrum reference annotated at peptide identifiction!");
     }
-    const MSSpectrum& exp_spectrum = exp[map_to_spectrum.at(pep_id.getMetaValue("spectrum_reference").toString())];
+    const MSSpectrum& exp_spectrum = exp[map_to_spectrum.at(pep_id.getSpectrumReference().toString())];
```
Comment (jpfeuffer) [inline src/openms/source/QC/FragmentMassError.cpp]:
```suggestion
    const MSSpectrum& exp_spectrum = exp[map_to_spectrum.at(pep_id.getSpectrumReference())];
```

File: src/openms/include/OpenMS/METADATA/PeptideIdentification.h [hunk 2]
```diff
@@ -145,6 +145,15 @@ namespace OpenMS
     /// sets the experiment label for this identification
     void setExperimentLabel(const String& type);
 
+    /// returns the spectrum reference for this identification. Currently it should
+    /// almost always be the full native vendor ID.
+    // TODO make a mandatory data member, add to idXML schema, think about storing the
+    //  extracted spectrum "number" only!
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/METADATA/PeptideIdentification.h:151]: We might need to Consider that there might be multiple spectrum=1 if different detectors are used etc. we recently had a similar issue

Comment (jpfeuffer) [reply src/openms/include/OpenMS/METADATA/PeptideIdentification.h:151]: Ok then I think one should raise an issue at sage/percolator regarding the format.

File: src/openms/source/METADATA/PeptideIdentification.cpp [hunk 3]
```diff
@@ -177,6 +177,16 @@ namespace OpenMS
     id_ = id;
   }
 
+  String PeptideIdentification::getSpectrumReference() const
```
Comment (timosachsenberg) [inline src/openms/source/METADATA/PeptideIdentification.cpp:180]: @hendrikweisser something that might be useful for the ID datastructure as well

Comment (hendrikweisser) [reply src/openms/source/METADATA/PeptideIdentification.cpp:180]: Can you elaborate?
The new ID data structure has a more explicit representation of a PSM (https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/METADATA/ID/ObservationMatch.h) with links to the the peptide and spectrum (https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/METADATA/ID/Observation.h), where the relevant metadata is stored.

**Pull request merged**